src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/Eventing/ParameterCapturingEventSource.cs

```diff
@@ -82,6 +83,9 @@ public void FailedToCapture(Guid RequestId, Exception ex)
         [Event(ParameterCapturingEvents.EventIds.Flush)]
         protected override void Flush()
         {
+            // This method is called on a timer and we shouldn't track any calls inside so we don't flood the user with unimportant captures.
+            using IDisposable _ = MonitorExecutionContextTracker.MonitorScope();
```

Comment:
Let's add a test to enforce validate this new behavior.

This unit test should give a pretty good example of where / how to test it.:

https://github.com/dotnet/dotnet-monitor/blob/d7bb04ff9bba038c37d67bde9dfb5d95331e7f99/src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/FunctionProbes/FunctionProbesScenario.cs#L320-L333


You might need to update the StartupHook project's `InternalsVisibleTo` to include the `UnitTestApp` project and you'll want to add a new entry around here to get the test to run: https://github.com/dotnet/dotnet-monitor/blob/d7bb04ff9bba038c37d67bde9dfb5d95331e7f99/src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/FunctionProbes/FunctionProbesScenario.cs#L57

Reply:
Looks like I commented on the wrong file, I meant to comment on FunctionprobesStub.cs